src/main/java/org/jenkins_ci/plugins/flexible_publish/builder/FailAtEndBuilder.java

```diff
@@ -0,0 +1,176 @@
+            throws InterruptedException, IOException {
+        // do as AbstractBuild.AbstractRunner#performAllBuildSteps
+        boolean wholeResult = true;
+        for (BuildStep buildstep: buildsteps) {
+            try {
+                if (!buildstep.perform(build, launcher, listener)) {
+                    listener.error(String.format(
+                            "[flexible-publish] %s failed",
+                            FlexiblePublisher.getBuildStepDetailedName(buildstep)
+                    ));
+                    build.setResult(Result.FAILURE);
+                    wholeResult = false;
+                }
+            } catch (Exception e) {
+                e.printStackTrace(listener.error(String.format(
+                        "[flexible-publish] %s aborted due to exception",
+                        FlexiblePublisher.getBuildStepDetailedName(buildstep)
+                )));
+                build.setResult(Result.FAILURE);
+                wholeResult = false;
+            }
+        }
+        return wholeResult;
+    }
+    
+    @Override
+    public Descriptor<Builder> getDescriptor() {
+        return new Descriptor<Builder>() {
+            @Override
+            public String getDisplayName() {
+                return String.format("[%s]", StringUtils.join(
+                        Lists.transform(
+                                buildsteps,
+                                new Function<BuildStep, String>() {
+                                    @Override
+                                    public String apply(BuildStep input) {
+                                        if (input instanceof Describable) {
+                                            return ((Describable<?>)input).getDescriptor().getDisplayName();
+                                        }
+                                        return input.getClass().getName();
+                                    }
+                                }
+                        ),
+                        ", "
+                ));
+            }
+        };
+    }
+    
+    /**
+     * Not Supported
+     * 
+     * @param project
+     * @return
+     * @deprecated
+     * @see hudson.tasks.BuildStep#getProjectAction(hudson.model.AbstractProject)
+     */
+    @Override
+    public Action getProjectAction(AbstractProject<?, ?> project) {
+        throw new UnsupportedOperationException();
+    }
+    
+    /**
+     * @param project
+     * @return
+     * @see hudson.tasks.BuildStep#getProjectActions(hudson.model.AbstractProject)
+     */
+    @Override
+    public Collection<? extends Action> getProjectActions(AbstractProject<?, ?> project) {
+        throw new UnsupportedOperationException();
+    }
+    
+    /**
+     * @return
+     * @see hudson.tasks.BuildStep#getRequiredMonitorService()
+     */
+    @Override
+    public BuildStepMonitor getRequiredMonitorService() {
+        throw new UnsupportedOperationException();

```

Comment:
<code>BuildStepMonitor.NONE</code>?
Is there any real need to throw runtime exceptions on all three calls?